[WIP] Use terraform to run CI

janisz opened the pull request.

janisz committed:
Use terraform to run CI

makkes commented on scripts/launch_aws_cluster.sh:
```diff
@@ -0,0 +1,11 @@
+
+bash <(curl -s https://raw.githubusercontent.com/dcos/dcos/master/test_util/terraform_init.sh)
+
+./terraform init
```
If I understand this line correctly, this is already done here:

https://github.com/dcos/dcos/blob/09608a1c0e0bf2237ffd0afb1111faa3414a6abb/test_util/terraform_init.sh#L79

janisz committed:
Remove unecesary terafform init

janisz committed:
Fix main.tf escaping

makkes commented on scripts/launch_aws_cluster.sh:
```diff
@@ -0,0 +1,10 @@
+
+bash <(curl -s https://raw.githubusercontent.com/dcos/dcos/master/test_util/terraform_init.sh)
+
+./terraform apply
```
```suggestion
./terraform apply --auto-approve
```

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
Be quiet:

janisz committed:
Auto approve

janisz committed:
Add debug

janisz committed:
Install jq

janisz committed:
Celanup cluster

janisz committed:
Use  as a ssh key

janisz committed:
No color

janisz committed:
Revert telegraf change

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
Use env vars instead of CLI args for terraform

janisz committed:
Use script insteadof envvronmet

janisz committed:
fix CWD

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
Fix typo

janisz committed:
SKip plan

janisz committed:
Generate id_rsa.pub

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
ssh-add

janisz committed:
ssh-add

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
WIP

janisz committed:
Use docker

janisz committed:
Revert "Use docker"

This reverts commit 2bd420ff249a00af7921b5f148e544e71ba4f1e7.

janisz committed:
WIP

janisz committed:
Cd to scripts

https://github.com/terraform-providers/terraform-provider-aws/issues/4275#issuecomment-391410544

janisz committed:
Chdir

janisz committed:
Use mesos

janisz committed:
WIP

janisz committed:
Fix

janisz committed:
Use docker

janisz committed:
WIP

janisz committed:
Do not generate a key

janisz committed:
WIP

drewvanstone commented on scripts/launch_aws_cluster.sh:
```diff
@@ -0,0 +1,70 @@
+    CHECKSUM_terraform="9bf9567bb0d396f83dd8ee2cedde5f188bdbba3316528648798e1d3f6814a1f7"
+    CHECKSUM_terraform_provider_aws="bdb07595b393c2fba7e0a679e1d9355e9b30edeec138f48fd1ff9deb2065a8f9"
+else
+    # How did you even get here?
```
😄

bamarni replied:
What we are currently doing for Windows is spin-up the cluster from a Linux machine and then storing the master node IP in a variable which is read from the Windows node:

https://github.com/dcos/dcos-core-cli/blob/2bc9f1b3f61fe50a4c863ae2f176c30815d473b8/ci/integration-tests-windows.groovy#L47-L56

This is not ideal but was done like this because dcos-launch doesn't run on Windows, it for example forced us to use the scripted pipeline instead of the declarative pipeline like other platforms.

On Windows, if we can spin-up the cluster from the node running the tests directly without having this multi-node steps, I think that's preferable. At least we seem to publish a binary for Windows as well (https://downloads.mesosphere.com/dcos-terraform/terraform/v0.11.13-mesosphere/windows_amd64/terraform.exe).

janisz replied:
Take a look at https://github.com/dcos/dcos-core-cli/pull/387 it's the first step to unify windows Mac and Linux integration tests.
